File: hw/jacome/U3/HW18Singleton/JS/view/SingletonPattern.js
const Calculator = require('../controller/Calculator');
const USTax = require('../model/USTax');

const calculator = new Calculator();

let subtotal = 10.0;
let total = calculator.salesTotal(subtotal);
console.log("The total is --> $" + total);

USTax.getInstance().setTaxPercentage(0.15);
total = calculator.salesTotal(subtotal);
console.log("The total with 15% tax is --> $" + total);

File: hw/jacome/U3/HW18Singleton/HW18Singleton/SingletonPattern/src/main/java/ec/edu/espe/singletonpattern/view/SingletonPattern.java
package ec.edu.espe.singletonpattern.view;

import ec.edu.espe.singletonpattern.controller.Calculator;
import ec.edu.espe.singletonpattern.model.USTax;

/**
 *
 * @author Micaela Jacome DESKTOP-46VMNHU ESPE
 */
public class SingletonPattern {

    public static void main(String[] args) {
        Calculator calculator = new Calculator();

        float subtotal = 10.0F;
        float total = calculator.salesTotal(subtotal);
        System.out.println("The total is --> $" + total);

        USTax.getInstance().setTaxPercentage(0.15F);
        total = calculator.salesTotal(subtotal);
        System.out.println("The total with 15% tax is --> $" + total);
    }
}
